integrations/access/common/recipient.go

```diff
@@ -150,6 +150,15 @@ func (s *RecipientSet) ToSlice() []Recipient {
 	return recipientSlice
 }
 
+// GetNames returns a slice of the recipient names in the set.
+func (s *RecipientSet) GetNames() []string {
+	names := make([]string, 0, len(s.recipients))
+	for _, recipient := range s.recipients {
+		names = append(names, recipient.Name)
+	}
+	return names
+}
+
```

Comment:
I don't think this is needed, the data flow is the following:
- we are given receipient in a map[role]rawRecipient
- we get RawRecipients for a given role
- we convert those to receipients, which is a richer data structure that contains the original user-provided string, but als the recipient kindand the recipient identifier once resolved
- we store those recipients in a set to avoid duplication (hugo.hervieux[at]goteleport.com and [at]hugo might resolve to the same user)
- we use those recipients in every message-sending method

This function is the equivalent of extracting RawRecipeints from and already parsed/resolved recipients in a deduplicated RecipienSet. This is going backward and should not happen as we'll need to parse/resolve them back again down the line. Either downstream function should take a `Recipient` instead of a RawRecpient, or the RecpientSet was not the correct way to store the user-proivided string and it should be a `RawRecipient`.

In both cases I think we should remove this function as this should never happen and indicates something is wrong with the recipient handling.

Reply:
I agree that the data glow is off here. How about something like this where the AMR cache just returns the raw recipients instead of resolving them and now they are resolved further down the chain in the same way the existing ones were?

Reply:
> How about something like this where the AMR cache just returns the raw recipients
Sounds good. My main question is why do we need the GetNames() in the msteams plugin PR while we have existing hosted plugins using AMRs to pick recipients?

Is the teams plugin doing something specific compared to the existing slack hosted plugin? How is the slack plugin doing this? Dumping all recipients and taking their names manually?

Reply:
The slack plugin plugin uses the shared accessrequest/app.go app isntead of a custom one, so has a getMessageRecipients that recieves recipients from the AMR cache which is passed the underlying bot's fetchrecipients as a callback. The msteams plugin behaves differently behaves differently in that it has its own recipient type and caches them as well. We could rewrite the msteams plugin to use the commmon recipient type but i think that is out of the scope of this PR

Reply:
I can make a follow up pr to address this